deepchem/models/dft/nnxc.py

```diff
@@ -0,0 +1,152 @@
+from abc import abstractproperty, abstractmethod
+from typing import Union
+try:
+    from dqc.xc.base_xc import BaseXC
+    from dqc.utils.datastruct import ValGrad, SpinParam
+    from dqc.api.getxc import get_xc
+    import torch
+except ModuleNotFoundError:
+    raise ModuleNotFoundError("This layer requires dqc and torch")
+
+
+class BaseNNXC(BaseXC, torch.nn.Module):
+    """
+    Base class for the NNLDA and HybridXC classes.
+    """
+
+    @abstractproperty
+    def family(self) -> int:
+        pass
+
+    @abstractmethod
+    def get_edensityxc(
+            self, densinfo: Union[ValGrad, SpinParam[ValGrad]]) -> torch.Tensor:
+        pass
+
+
+class NNLDA(BaseNNXC):
+    """
+    Neural network xc functional of LDA (only receives the density as input)
+    """
+
+    def __init__(self,
+                 nnmodel: torch.nn.Module,
+                 ninpmode: int = 1,
+                 outmultmode: int = 1):
+        super().__init__()
+        """
+        Parameters
+        ----------
+        nnmodel: torch.nn.Module
+            Neural network for xc functional
+        ninpmode: int
+            The mode to decide the transformation of the density to NN input.
+        outmultmode: int
+            The mode to decide Eks from NN output
+        """
+        self.nnmodel = nnmodel
+        self.ninpmode = ninpmode
+        self.outmultmode = outmultmode
+
+    @property
+    def family(self) -> int:
+        return 1
+
+    def get_edensityxc(
+            self, densinfo: Union[ValGrad, SpinParam[ValGrad]]) -> torch.Tensor:
+        """
+        This method transform the local electron density (n) and the spin
+        density (xi) for polarized and unpolarized cases, to be the input of
+        the neural network.
+
+        Parameters
+        ----------
+        densinfo: Union[ValGrad, SpinParam[ValGrad]]
+            Density information calculated using DQC utilities.
+        Returns
+        -------
+        res
+            Neural network output by calculating total density (n) and the spin density (xi)
+        """
+        if isinstance(densinfo, ValGrad):  # unpolarized case
+            n = densinfo.value.unsqueeze(-1)  # (*BD, nr, 1)
+            xi = torch.zeros_like(n)
+        else:  # polarized case
+            nu = densinfo.u.value.unsqueeze(-1)
+            nd = densinfo.d.value.unsqueeze(-1)
+            n = nu + nd  # (*BD, nr, 1)
+            xi = (nu - nd) / (n + 1e-18)  # avoiding nan
+
+        ninp = n
+
+        x = torch.cat((ninp, xi), dim=-1)  # (*BD, nr, 2)
+        nnout = self.nnmodel(x)  # (*BD, nr, 1)
+        res = nnout * n  # (*BD, nr, 1)
+        res = res.squeeze(-1)
+        return res
+
+
+class HybridXC(BaseNNXC):
+    """
+    The HybridXC module computes XC energy by summing XC energy computed
```

Comment:
Can you add a usage example? Can you also extend the explanation of what this layer does?